packages/front-end/components/Settings/EditDataSource/FeatureEvaluationQueries/FeatureEvaluationQueries.tsx

```diff
@@ -0,0 +1,273 @@
+  );
+
+  const handleSave = useCallback(
+    (idx: number) => async (featureUsageQuery: FeatureUsageQuery) => {
+      const copy = cloneDeep<DataSourceInterfaceWithParams>(dataSource);
+      // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
+      if (!copy.settings.queries.featureUsage) {
+        // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
+        copy.settings.queries.featureUsage = [];
+      }
+      // @ts-expect-error TS(2532) If you come across this, please fix it!: Object is possibly 'undefined'.
+      copy.settings.queries.featureUsage[idx] = featureUsageQuery;
+      await onSave(copy);
+    },
+    [dataSource, onSave],
+  );
+
+  const [validatingQuery, setValidatingQuery] = useState(false);
+
+  const handleValidate = useCallback(
+    () => async () => {
+      const copy = cloneDeep<DataSourceInterfaceWithParams>(dataSource);
+      setValidatingQuery(true);
+      // Resaving the document as-is will automatically revalidate any queries in error state
+      await onSave(copy);
+      setValidatingQuery(false);
+    },
+    [dataSource, onSave],
+  );
+
+  if (!dataSource) {
+    console.error("ImplementationError: dataSource cannot be null");
+    return null;
+  }
+
+  return (
+    <Box>
+      <Flex align="center" gap="2" mb="3" justify="between">
+        <Box>
+          <Flex align="center" gap="3" mb="0">
+            <Heading as="h3" size="4" mb="0">
+              Feature Usage Query
+            </Heading>
+          </Flex>
+        </Box>
+
+        {canEdit && featureUsageQueries.length === 0 && (
+          <Box>
+            <Button onClick={handleAdd}>
+              <FaPlus className="mr-1" /> Add
+            </Button>
+          </Box>
+        )}
+      </Flex>
+      <p>
+        Returns a list of feature evaluation events for feature evaluation
+        diagnostics.
+      </p>
+
+      {/* region Empty state */}
+      {featureUsageQueries.length === 0 ? (
+        <Callout status="info">
+          A feature usage query has not been added. Feature usage queries are
+          required for feature evaluation diagnostics.
+        </Callout>
+      ) : null}
+      {/* endregion Empty state */}
+
+      {featureUsageQueries.map((query, idx) => {
+        const isOpen = openIndexes[idx] || false;
+
+        return (
+          <Card mt="3" key={query.id}>
+            <Flex align="start" justify="between" py="2" px="3" gap="3">
+              {/* region Title Bar */}
+              <Box width="100%">
+                <Flex>
+                  <Heading as="h4" size="3" mb="1">
+                    {query.name}
+                  </Heading>
+                  {query.description && (
+                    <p className="ml-3 text-muted">{query.description}</p>
+                  )}
+                </Flex>
+
+                <Flex gap="4"></Flex>
+                {query.error && (
```

Comment:
question: I was not able to show this. Where is this being set / when would this appear?